BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/CoreEventSource.cs

```diff
@@ -618,6 +629,15 @@ public void InitializationIsSkippedForSampledItem(string appDomainName = "Incorr
         [Event(66, Message = "Call to WindowsIdentity.Current failed with the exception: {0}.", Level = EventLevel.Warning)]
         public void LogWindowsIdentityAccessSecurityException(string error, string appDomainName = "Incorrect") => this.WriteEvent(66, error ?? string.Empty, this.nameProvider.Name);
 
+        [Event(67, Message = "Backend has responded with {0} status code in {1}ms.", Level = EventLevel.Warning)]
+        public void BreezeResponseTime(int responseCode, float responseDurationInMs, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(67, responseCode, responseDurationInMs, this.nameProvider.Name);
+#if NETSTANDARD2_0
+            this.BreezeResponseTimeCounter.WriteMetric(responseDurationInMs);
```

Comment:
Should we have separate method for the WriteEvent and WriteMetric? So that we can subscribe to just the EventCounter part, without paying the penality for the WriteEvent?

Reply:
Sure. This is a very good recommendation. But we may need to make two logging calls like below. One for events and another for counter. 

CoreEventSource.Log.BreezeResponseTime(response != null ? (int)response.StatusCode : -1, responseDuration);
CoreEventSource.Log.BreezeResponseTimeCounter(responseDuration);

When we run the collection for event counters, it collects only counter metrics and not event data.  
dotnet-trace collect --process-id 30280 --providers Microsoft-ApplicationInsights-Core:0:1:EventCounterIntervalSec=1  

We may need to either call log two times in few places or utilize WriteEvent to have a counter. Could you please advise which is better?

Reply:
As per our discussion, creating a nonevent with event source class and calling log twice from caller.